src/options.c

```diff
@@ -369,30 +399,86 @@ JOHN_USAGE_REGEX \
 "--node=MIN[-MAX]/TOTAL     this node's number range out of TOTAL count\n" \
+"--format=CLASS             valid classes: dynamic, cpu\n" \
+"--subformat=FORMAT         pick a benchmark format for --format=crypt\n" \
+"--mkpc=N                   request a lower max. keys per crypt\n" \
+"--min-length=N             request a minimum candidate length in bytes\n" \
+"--max-length=N             request a maximum candidate length in bytes\n" \
+"--length=N                 shortcut for --min-len=N --max-len=N\n" \
+"--field-separator-char=C   use 'C' instead of the ':' in input and pot files\n" \
+"--fix-state-delay=N        performance tweak, see doc/OPTIONS\n" \
+"--log-stderr               log to screen instead of file\n"             \
+"--verbosity=N              change verbosity (1-%u or %u for debug, default %u)\n" \
+"--no-log                   disables creation and writing to john.log file\n"      \
+"--bare-always-valid=Y      treat bare hashes as valid (Y/N)\n" \
+"--progress-every=N         emit a status line every N seconds\n" \
+"--crack-status             emit a status line whenever a password is cracked\n" \
+"--keep-guessing            try finding plaintext collisions\n" \
+"--no-keep-guessing         turn off any default \"keep guessing\" for used format\n" \
+"--max-candidates=[-]N      gracefully exit after this many candidates tried.\n" \
+"                           (if negative, reset count on each crack)\n" \
+"--max-run-time=[-]N        gracefully exit after this many seconds (if negative,\n" \
+"                           reset timer on each crack)\n" \
+"--regen-lost-salts=N       brute force unknown salts (see doc/OPTIONS)\n" \
+"--rules-skip-nop           skip any NOP \":\" rules (you already ran w/o rules)\n" \
+"--mkv-stats=FILE           \"markov\" stats file (see doc/MARKOV)\n" \
+"--reject-printable         reject printable binaries\n" \
+"--show=formats             show information about hashes in a file (JSON)\n" \
+"--show=invalid             show lines that are not valid for selected format(s)\n" \
+"--skip-self-tests          skip self tests\n" \
+"--test-full=LEVEL          run more thorough self-tests\n" \
+"--stress-test[=TIME]       loop self tests forever\n" \
+"--no-mask                  used with --test for alternate benchmark w/o mask\n" \
 JOHN_USAGE_FORK \
 "--pot=NAME                 pot file to use\n" \
-"--list=WHAT                list capabilities, see --list=help or doc/OPTIONS\n"
+"--list=WHAT                list capabilities, see --list=help or doc/OPTIONS\n" \
+"--input-encoding=NAME      input encoding (alias for --encoding)\n" \
+"--internal-codepage=NAME   codepage used in rules/masks (see doc/ENCODINGS)\n" \
+"--target-encoding=NAME     output encoding (used by format, see doc/ENCODINGS)\n" \
+"--tune=HOW                 tuning options (auto/report/N)\n" \
+"--incremental-charcount=N  override CharCount for incremental mode\n" \
```

Comment:
--incremental-charcount should go right after `--incremental`

Reply:
Applied.